### PR TITLE
Update pipeline version to 6 and pin versions

### LINKED ISSUE #2262: Handle pipeline migration concerns for BYO catalog type changes
**Is your feature request related to a problem? Please describe.**
We will have another 1:1 node `op` change for our preloaded components that will require some frontend pipeline migration and migration of our pytest resources.

- [ ] Frontend migration
- [x] Resource updates (completed in #2241) 

**Describe the solution you'd like**
The following 1:1 mapping should work for the built-in catalog components:
```
"run_notebook_using_papermill_Runnotebookusingpapermill": "local-directory-catalog:61e6f4141f65",
"filter_text_using_shell_and_grep_Filtertext": "local-directory-catalog:737915b826e9",
"component_Downloaddata": "url-catalog:c6c0588048ae",
"component_Calculatedatahash": "url-catalog:4fc759382b1b",
"bash_operator_BashOperator": "url-catalog:49f8e61b78c3",
"email_operator_EmailOperator": "url-catalog:8bef428ea3cd",
"http_operator_SimpleHttpOperator": "url-catalog:e97030fb448a",
"spark_sql_operator_SparkSqlOperator": "url-catalog:ff0d51b70719",
"spark_submit_operator_SparkSubmitOperator": "url-catalog:2756314f3ff5",
"slack_operator_SlackAPIPostOperator": "local-file-catalog:81b4f925702e"
```

We should also have the following mapping for the `component_source` property for the properties below (`old -> new`):

- `"component_Calculatedatahash/"url-catalog:4fc759382b1b"`:

```
'https://raw.githubusercontent.com/kubeflow/pipelines/1.6.0/components/basics/Calculate_hash/component.yaml'
->
{'catalog_type': 'url-catalog', 'component_ref': {'url': 'https://raw.githubusercontent.com/kubeflow/pipelines/1.6.0/components/basics/Calculate_hash/component.yaml'}}
```

- `"component_Downloaddata"/"url-catalog:c6c0588048ae"`

```
'https://raw.githubusercontent.com/kubeflow/pipelines/1.6.0/components/web/Download/component.yaml'
->
{'catalog_type': 'url-catalog', 'component_ref': {'url': 'https://raw.githubusercontent.com/kubeflow/pipelines/1.6.0/components/web/Download/component.yaml'}}
```

- `"bash_operator_BashOperator"/"url-catalog:49f8e61b78c3"`:

```
'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/operators/bash_operator.py'
->
{'catalog_type': 'url-catalog', 'component_ref': {'url': 'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/operators/bash_operator.py'}}
```

- `"email_operator_EmailOperator"/"url-catalog:8bef428ea3cd"`:

```
'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/operators/email_operator.py'
->
{'catalog_type': 'url-catalog', 'component_ref': {'url': 'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/operators/email_operator.py'}}
```

- `"http_operator_SimpleHttpOperator"/"url-catalog:e97030fb448a"`:

```
'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/operators/http_operator.py'
->
{'catalog_type': 'url-catalog', 'component_ref': {'url': 'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/operators/http_operator.py'}}
```

- `"spark_sql_operator_SparkSqlOperator"/"url-catalog:ff0d51b70719"`:

```
'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/contrib/operators/spark_sql_operator.py'
->
{'catalog_type': 'url-catalog', 'component_ref': {'url': 'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/contrib/operators/spark_sql_operator.py'}}
```

- `"spark_submit_operator_SparkSubmitOperator"/"url-catalog:2756314f3ff5"`:

```
'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/contrib/operators/spark_submit_operator.py'
->
{'catalog_type': 'url-catalog', 'component_ref': {'url': 'https://raw.githubusercontent.com/apache/airflow/1.10.15/airflow/contrib/operators/spark_submit_operator.py'}}
```

The remaining 3 components are a bit stickier in that the old and new values include absolute locations on a user's machine. The general format for these components will be as follows:

- `"run_notebook_using_papermill_Runnotebookusingpapermill"/"local-directory-catalog:61e6f4141f65"`:

```
'[ENV_JUPYTER_PATH[0]]/components/kfp/run_notebook_using_papermill.yaml'
->
{'catalog_type': 'local-directory-catalog', 'component_ref': {'base_dir': '[ENV_JUPYTER_PATH[0]]/components/kfp', 'path': 'run_notebook_using_papermill.yaml'}}
```

- `"filter_text_using_shell_and_grep_Filtertext"/"local-directory-catalog:737915b826e9"`:

```
'[ENV_JUPYTER_PATH[0]]/components/kfp/filter_text_using_shell_and_grep.yaml'
->
{'catalog_type': 'local-directory-catalog', 'component_ref': {'base_dir': '[ENV_JUPYTER_PATH[0]]/components/kfp', 'path': 'filter_text_using_shell_and_grep.yaml'}}
```

- `"slack_operator_SlackAPIPostOperator"/"local-file-catalog:81b4f925702e"`:

```
'[ENV_JUPYTER_PATH[0]]/components/airflow/slack_operator.py'
->
{'catalog_type': 'local-file-catalog', 'component_ref': {'base_dir': '[ENV_JUPYTER_PATH[0]]/components', 'path': 'airflow/slack_operator.py''}}
```

**Additional context**
Since this is a relatively easy change, I would probably hold off on implementing it until we're sure that we've finalized these `op`s.

### PR DESCRIPTION
**Pull request by ajbozarth**
Fixes #2262

Sister PR to https://github.com/elyra-ai/pipeline-editor/pull/172

Update the pipeline version and migration code to handle the changes
in Elyra 3.3

Pin package versions to prevent future backwards compatibility errors

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ajbozarth** [comment]
Leaving in draft until pipeline-editor is released and I update the version. I've updated it to point at the code sandbox for easier testing

**Comment by ptitzler** [comment]
Tested migration using pipelines from the examples repository:

Without example components _installed_:
- [X] For pipelines containing custom components migration "fails" as expected -> might want to tweak title wording and use "aborted", if possible
- [X] For pipelines containing only generic components migration succeeds (and yield the expected run results)

Without example components _enabled_:
- [X] Migration "fails" as expected -> might want to tweak title wording and use "aborted", if possible

With example components installed:
- [X] examples/pipelines/kubeflow_pipelines_component_examples (all pipelines) 
- [X] examples/pipelines/airflow_component_examples/bash_operator
- [ ] examples/pipelines/airflow_component_examples/email_operator
- [ ] examples/pipelines/airflow_component_examples/http_operator
- [ ] examples/pipelines/airflow_component_examples/slack_api_post_operator 
- [ ] examples/pipelines/airflow_component_examples/spark_sql_operator 
- [ ] examples/pipelines/airflow_component_examples/spark_submit_operator
- [ ] examples/pipelines/

**Comment by ptitzler** [comment]
Migrate and run https://github.com/CODAIT/flight-delay-notebooks
- [ ] on KFP
- [ ] on AA

**Comment by ptitzler** [comment]
Migrate and run https://github.com/CODAIT/covid-notebooks

- [ ] on KFP
- [ ] on AA

**Comment by ajbozarth** [comment]
I just pushed an update to the code sandbox version with the latest fix from the pipeline editor PR. This includes bug fixes for all reported issues. I'll be working on addressing the tests next

**Comment by akchinSTC** [comment]
> * examples/pipelines/airflow_component_examples/spark_sql_operator*
> * examples/pipelines/airflow_component_examples/spark_submit_operator*

I wouldnt be opposed to removing these two examples from the repo given the amount of setup these require. 
Currently fixing the slack integration connection on the test cluster to test.

Successfully migrated my AA and kfp 3.1.x and 3.2.x pipelines with these latest changes

**Review by marthacryan — COMMENTED** [review]
About to take a look at the code now but I just tried this out with a couple of older pipelines and it seemed to be working well.

**Review by ptitzler — APPROVED** [review]
LGTM

**Review by kiersten-stokes — APPROVED** [review]
LGTM!

File: packages/pipeline-editor/src/PipelineEditorWidget.tsx [hunk 1]
```diff
@@ -349,44 +354,87 @@ const PipelineWrapper: React.FC<IProps> = ({
...
+                      contextRef.current.path,
+                      node.app_data.filename
+                    );
+                  }
+                  return pipeline;
+                },
+                migrationPalette
+              );
+              contextRef.current.model.fromString(
+                JSON.stringify(migratedPipeline, null, 2)
```
Comment (marthacryan) [inline packages/pipeline-editor/src/PipelineEditorWidget.tsx:385]: Why are you converting this to a string before saving it? Couldn't you just use `model.fromJSON`?

Comment (ajbozarth) [reply packages/pipeline-editor/src/PipelineEditorWidget.tsx:385]: This step prettifies the json

**Review by marthacryan — APPROVED** [review]

**Review by akchinSTC — APPROVED** [review]

**Pull request merged**